### PR TITLE
refactored secret script

### PR DESCRIPTION
**Pull request by Climbatize**
# Description
Duplicated changes from MiniApp repo secret script to this one

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [X] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

File: configure-secrets.sh [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+usage () {
+cat <<HELP_USAGE
+usage: $0 param1 param[n]...
+
+   Parameters:
+   - param1: The SDK name which needs to have a -Secrets.xcconfig file
+   - param[n]: a list of secrets environment variables to put in secret config file
+
+  Example usage:
+    $0 MiniApp RMA_API_ENDPOINT RAS_PROJECT_SUBSCRIPTION_KEY RAS_PROJECT_IDENTIFIER
```
Comment (imaginaris) [inline configure-secrets.sh]: I think this example doesn't match InAppMessaging :)

**Review by imaginaris — CHANGES_REQUESTED** [review]
Great improvement 👍 
Just one comment to address.

File: configure-secrets.sh (hunk 1 above)
Comment (Climbatize) [reply configure-secrets.sh]: I don't see what's wrong 😚

Comment (imaginaris) [reply configure-secrets.sh]: Shouldn't that be `$0 InAppMessaging RIAM_CONFIG_URL...` instead of `$0 MiniApp...`?

Comment (Climbatize) [reply configure-secrets.sh]: I was a joke, sorry XD
Sometimes I should precise ^^

Comment (Climbatize) [reply configure-secrets.sh]: done

**Review by imaginaris — APPROVED** [review]

File: configure-secrets.sh [hunk 2]
```diff
@@ -40,8 +48,36 @@ echo "//" >> $SECRETS_FILE
...
+  cmd='echo "${var_name} = $SECURE_DOUBLE_SLASH_FOR_XCCONFIG_RESULT"'
+  eval ${cmd} >> $SECRETS_FILE
+done
 
-# Set secrets from environment variables
-echo "RIAM_CONFIG_URL = $CONFIG_URL" >> $SECRETS_FILE
-echo "RIAM_APP_SUBSCRIPTION_KEY = $SUBSCRIPTION_KEY" >> $SECRETS_FILE
\ No newline at end of file
+if [ $success = false ] ; then
+	echo "Before building the project you must set missing environment variables. See project README for instructions."
```
Comment (donnie-jp) [inline configure-secrets.sh:82]: @Climbatize 👍 
I think the README might need updated too.

Comment (imaginaris) [reply configure-secrets.sh:82]: But there are already some instructions in README to set up env variables for Test app.

**Pull request merged**